### PR TITLE
Obsolete old APIs

### PR DESCRIPTION
**Pull request by BrennanConroy**

### CODE REVIEW

**Comment by cwe1ss** [comment]
Awesome, it's great to see this!

Small nitpick: There's a Visual Studio Version downgrade in the sln file.

File: src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs [hunk 1]
```diff
@@ -19,9 +19,15 @@ public interface ILoggerFactory : IDisposable
         ILogger CreateLogger(string categoryName);
 
         /// <summary>
+        /// <para>
+        /// This method is being removed from the ILoggerFactory abstraction, to have similar behavior
```
Comment (Eilon) [inline src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs]: Recommend following a pattern closer to the guidelines: https://github.com/aspnet/Home/wiki/Engineering-guidelines#breaking-changes

File: src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs [hunk 2]
```diff
@@ -19,9 +19,15 @@ public interface ILoggerFactory : IDisposable
...
+        /// use the concrete LoggerFactory class with Microsoft.Extensions.Logging.LoggerFactory.AddProvider(string, ILoggerProvider)
+        /// </para>
         /// Adds an <see cref="ILoggerProvider"/> to the logging system.
         /// </summary>
         /// <param name="provider">The <see cref="ILoggerProvider"/>.</param>
+        [Obsolete("This method is being removed from the ILoggerFactory abstraction, to have similar behavior" +
```
Comment (Eilon) [inline src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs]: Ditto.

File: src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs [hunk 3]
```diff
@@ -19,9 +19,13 @@ public interface ILoggerFactory : IDisposable
         ILogger CreateLogger(string categoryName);
 
         /// <summary>
+        /// <para>
+        /// This method is obsolete, use AddProvider on the concrete Microsoft.Extensions.Logging.LoggerFactory class.
```
Comment (Eilon) [inline src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs]: Can we follow the guidelines on Recommend following a pattern closer to the guidelines: https://github.com/aspnet/Home/wiki/Engineering-guidelines#breaking-changes a little more closely? I recommend using the exact phrasing in there, i.e. `This method is obsolete and will be removed in a future version. The recommended alternative is Microsoft.Extensions.Logging.LoggerFactory.AddProvider().`

Comment (Eilon) [inline src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs]: (Applies to all the obsoletions in this PR.)

Comment (BrennanConroy) [inline src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs]: `This method is obsolete and will be removed in a future version. The recommended alternative is Microsoft.Extensions.Logging.AddConsole() on Microsoft.Extensions.Logging.LoggerFactory.`

Comment (BrennanConroy) [inline src/Microsoft.Extensions.Logging.Abstractions/ILoggerFactory.cs]: `This method is obsolete and will be removed in a future version. The recommended alternative is to call the Microsoft.Extensions.Logging.AddConsole() extension method on the Microsoft.Extensions.Logging.LoggerFactory instance.`

**Pull request merged**